cloudsync/event.py

```diff
@@ -88,18 +88,29 @@ def __init__(self, provider: "Provider", state: "SyncState", side: int,
         self.reauthenticate = reauth or self.__reauth
 
     def _validate_root(self):
-        if not self._root_validated and self.provider.connected:
-            (self._root_path, self._root_oid) = self.provider.set_root(self._root_path, self._root_oid)
+        if self._root_validated:
+            return True
+
+        if self.provider.root_path and self.provider.root_oid:
+            # provider's root set, use it
+            self._root_path = self.provider.root_path
+            self._root_oid = self.provider.root_oid
+
+        if not self._root_path and not self._root_oid:
+            # no root specified, nothing to validate
+            self._cursor_tag = self.label = "_cursor"
+            self.cursor = self.state.storage_get_data(self._cursor_tag)
+            self._root_validated = True
+
+        elif self._root_path and self._root_oid:
+            # path and oid specified, assume they are valid
             my_root = self._root_path or self._root_oid
-            if my_root:
```

Comment:
so, we no longer want to support unrooted/unwalked syncing?

Reply:
unrooted should still work -- I pulled the `else` (my_root is None) block below into its own `if` block (line 99 above) 

can't walk in the unrooted case since there is no path or oid to walk, but that has always been the case AFAICT.